src/persist-client/src/internal/gc.rs

```diff
@@ -260,7 +261,14 @@ where
                 break;
             }
         }
-        let state = states.into_inner();
+
+        // Delete the rollup blobs before removing them from state.
```

Comment:
The only thing I'm not yet getting is why we move this one around. It's order with respect to the truncate call is still the same.

Reply:
fixes an unrelated possible leak of rollup blobs if we crashed at the wrong time. before, we removed rollup from state and then deleted the blob, so if we crashed in between the two we'd leak it. now, we delete before removing from state, which is safe because the StateVersions code can't do anything useful with a rollup that is before the earliest live diff (like.. by definition)

Reply:
yep, that makes sense! Also makes sense why I didn't connect it with the real purpose of the PR. 😅

Reply:
that's my fault for not explaining it! tbh, I just forgot that i'd noticed and fixed this while I was in there 😳